multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/CreateServiceStep.java

```diff
@@ -84,6 +89,26 @@ private void processServiceCreationFailure(ProcessContext context, CloudServiceI
                              ExceptionMessageTailMapper.map(configuration, CloudComponents.SERVICE_BROKERS, serviceInstance.getLabel()));
     }
 
+    private void setServiceGuid(ProcessContext context, CloudControllerClient client, CloudServiceInstanceExtended serviceInstance) {
+        Set<DynamicResolvableParameter> dynamicResolvableParameters = context.getVariable(Variables.DYNAMIC_RESOLVABLE_PARAMETERS);
+        Optional<DynamicResolvableParameter> dynamicResolvableParameterOptional = dynamicResolvableParameters.stream()
+                                                                                                             .filter(dynamicParameter -> dynamicParameter.getParameterName()
+                                                                                                                                                         .equals(SupportedParameters.SERVICE_GUID)
+                                                                                                                 && dynamicParameter.getRelationshipEntityName()
+                                                                                                                                    .equals(serviceInstance.getResourceName()))
+                                                                                                             .findFirst();
+
+        if (dynamicResolvableParameterOptional.isPresent()) {
+            DynamicResolvableParameter dynamicResolvableParameter = dynamicResolvableParameterOptional.get();
+            String serviceInstanceGuid = client.getRequiredServiceInstanceGuid(serviceInstance.getName())
```

Comment:
do we need to make a new cf v3 call here to get the guid?

Reply:
Unfortunately yes, we need to make this additional call because [Create Service Instance](http://v3-apidocs.cloudfoundry.org/version/3.138.0/index.html#create-a-service-instance) request return async job to poll by default, only when service instance is user provided, service guid can be returned. That's why additional call will be made only on demand.